### PR TITLE
16/create sign in/sign up funtionality

### LINKED ISSUE #16: Create sign up/sign in funtionality
As a **user** I want to **be able to sign up and sign in on the website** so i can **have a profile and rent things and post listings**

**Acceptance criteria**

_Signup_
- [x] The user can choose an email
- [x] The user can choose a password
- [x] The username is stored in backend
- [x] The password is stored in backend (encrypted)

// will move this into validation issue
_Singin_
- [ ] The user cant sign in with the wrong email/password
- [x] The password is encrypted/hidden and never shown in plain text to the user

If we have time:
- [ ] The user should be able to reset the password
- [x] The user should be able to sign in and sign up with gmail
- [ ] The user should be able to sign in and sign up with facebook

### LINKED ISSUE #25: Create login / register page
As a user I want to be able to log in if I have an account or sign up if I dont

acceptance criteria:

_Signup_
- [ ] Sign up form should have inputs for email, firstname, lastname, password and password confirmation
- [ ] User should have the option to register with facebook or gmail

_Signin_
- [ ] inputs for email and password
- [ ] option to sign in with facebook or gmail

### PR DESCRIPTION
**Pull request by mirrenil**
This PR will:
- sign up new users
- sign in users with email and password
- allow users to sign up and sign in with google
- adds react-google-button

closes #16 
closes #25

### CODE REVIEW

File: src/Pages/SignIn.tsx [hunk 1]
```diff
@@ -1,11 +1,95 @@
...
+	const [error, setError] = useState("");
+	const [loading, setLoading] = useState(false);
+	const navigate = useNavigate();
+
+	const handleSubmit = async (e: FormEvent) => {
+		e.preventDefault();
+
+		try {
+		await login(loginEmail, loginPassword);
+		console.log("login successful");
```
Comment (liinneea-a) [inline src/Pages/SignIn.tsx:23]: Should we have this console log? Maybe we can keep it for now but remove it once we have a sign in succeeded confirmation

**Review by liinneea-a — APPROVED** [review]

File: src/Pages/SignIn.tsx (hunk 1 above)
Comment (mirrenil) [reply src/Pages/SignIn.tsx:23]: I will edit and add the password confirmation as well before we merge 🦘

Comment (liinneea-a) [reply src/Pages/SignIn.tsx:23]: password confirmation is already added! :)

Comment (mirrenil) [reply src/Pages/SignIn.tsx]: To firebase? I added it just now

**Review by hazem-89 — DISMISSED** [review]

**Review by hazem-89 — APPROVED** [review]

**Pull request merged**